### PR TITLE
Add Vercel deployment from PRs

### PR DESCRIPTION
**Pull request by safinsaf**
Add Vercel deployment from PRs

https://soramitsu.atlassian.net/jira/software/c/projects/DOPS/boards/43?modal=detail&selectedIssue=DOPS-2386&assignee=6163c8b178e5e40070bcd5f7

### CODE REVIEW

**Review by RamilMus — APPROVED** [review]

**Review by alexkg — DISMISSED** [review]

File: .github/workflows/vercel-deploy-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  vercel:
+    runs-on: ubuntu-latest
+    name: "Deploy application"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '16'
```
Comment (0x009922) [inline .github/workflows/vercel-deploy-dev.yaml:17]:
```suggestion
          node-version: '18'
```

File: .github/workflows/vercel-deploy-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        - '**'
+        - '!main'
+
+jobs:
+  vercel:
+    runs-on: ubuntu-latest
+    name: "Deploy application"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
```
Comment (0x009922) [inline .github/workflows/vercel-deploy-dev.yaml:15]:
```suggestion
      - uses: actions/setup-node@v3
```

**Comment by 6r1d** [comment]
LGTM generally, ping me for approval after applying the changes suggested by @0x009922.

File: .github/workflows/vercel-deploy-dev.yaml (hunk 2 above)
Comment (safinsaf) [reply .github/workflows/vercel-deploy-dev.yaml]: Done

File: .github/workflows/vercel-deploy-dev.yaml (hunk 1 above)
Comment (safinsaf) [reply .github/workflows/vercel-deploy-dev.yaml]: Done

**Review by 6r1d — APPROVED** [review]
LGTM

**Review by Cre-eD — APPROVED** [review]

File: .github/workflows/vercel-deploy-dev.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      branches:
+        - '**'
+        - '!main'
+
+jobs:
+  vercel:
+    runs-on: ubuntu-latest
+    name: "Deploy application"
+    steps:
+      - uses: actions/checkout@v2
```
Comment (wpi86) [inline .github/workflows/vercel-deploy-dev.yaml:14]: please, pin version

File: .github/workflows/vercel-deploy-dev.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/setup-node@v3
```
Comment (wpi86) [inline .github/workflows/vercel-deploy-dev.yaml:15]: please, pin version

**Review by wpi86 — CHANGES_REQUESTED** [review]

File: .github/workflows/vercel-deploy-dev.yaml (hunk 3 above)
Comment (safinsaf) [reply .github/workflows/vercel-deploy-dev.yaml]: Done

**Review by wpi86 — APPROVED** [review]

**Pull request merged**